ENIGMAsystem/SHELL/Graphics_Systems/None/fillin.h

```diff
@@ -0,0 +1,296 @@
+	gs_scalar texture_get_texel_height(int texid);
+	void texture_set_stage(int stage, int texid);
+	void texture_reset();
+	void texture_set_interpolation_ext(int sampler, bool enable);
+	void texture_set_repeat_ext(int sampler, bool repeat);
+	void texture_set_wrap_ext(int sampler, bool wrapu, bool wrapv, bool wrapw);
+	void texture_set_border_ext(int sampler, int r, int g, int b, double a);
+	void texture_set_filter_ext(int sampler, int filter);
+	void texture_set_lod_ext(int sampler, double minlod, double maxlod, int maxlevel);
+	bool texture_mipmapping_supported();
+	bool texture_anisotropy_supported();
+	float texture_anisotropy_maxlevel();
+	void texture_anisotropy_filter(int sampler, gs_scalar levels);
+
+	bool surface_is_supported();
+	int surface_create(int width, int height, bool depthbuffer);
+	int surface_create_msaa(int width, int height, int samples);
+	void surface_set_target(int id);
+	void surface_reset_target(void);
+	int surface_get_target();
+	void surface_free(int id);
+	bool surface_exists(int id);
+	int surface_get_texture(int id);
+	int surface_get_width(int id);
+	int surface_get_height(int id);
+	int surface_getpixel(int id, int x, int y);
+	int surface_getpixel_ext(int id, int x, int y);
+	int surface_getpixel_alpha(int id, int x, int y);
+
+	int surface_save(int id, string filename);
+
+	int surface_save_part(int id, string filename, unsigned x, unsigned y, unsigned w, unsigned h);
+	int background_create_from_surface(int id, int x, int y, int w, int h, bool removeback, bool smooth, bool preload);
+	int sprite_create_from_surface(int id, int x, int y, int w, int h, bool removeback, bool smooth, bool preload, int xorig, int yorig);
+	int sprite_create_from_surface(int id, int x, int y, int w, int h, bool removeback, bool smooth, int xorig, int yorig);
+	void sprite_add_from_surface(int ind, int id, int x, int y, int w, int h, bool removeback, bool smooth);
+	void surface_copy_part(int destination, gs_scalar x, gs_scalar y, int source, int xs, int ys, int ws, int hs);
+	void surface_copy(int destination, gs_scalar x, gs_scalar y, int source);
+
+	int draw_get_msaa_maxlevel();
+	bool draw_get_msaa_supported();
+	void draw_set_msaa_enabled(bool enable);
+	void draw_enable_alphablend(bool enable);
+	bool draw_get_alpha_test();
+	unsigned draw_get_alpha_test_ref_value();
+	void draw_set_alpha_test(bool enable);
+	void draw_set_alpha_test_ref_value(unsigned val);
+	void draw_set_line_pattern(int pattern, int scale);
+
+	int draw_getpixel(int x,int y);
+	int draw_getpixel_ext(int x,int y);
+
+	extern int window_get_region_height_scaled();
+
+	int sprite_create_from_screen(int x, int y, int w, int h, bool removeback, bool smooth, bool preload, int xorig, int yorig);
+	int sprite_create_from_screen(int x, int y, int w, int h, bool removeback, bool smooth, int xorig, int yorig);
+	void sprite_add_from_screen(int id, int x, int y, int w, int h, bool removeback, bool smooth);
+
+	int glsl_shader_create(int type);
```

Comment:
These sound GL specific... do we want to implement them in None? I suppose that's more of a philosophical question. These don't bother me.

Reply:
I kept this stuff under the impression that it was what was needed for shaders, but after looking over it again these are being wrapped to the general shader_\*() functions so I'll fill those in instead (also is it just me or is documentation on ENIGMA's shaders really sparse?)